### PR TITLE
DOCSUP-6549: Edited and translated to Russian

### PR DESCRIPTION
**Pull request by gyuton**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

Changelog category:
- Documentation for #19819, #19681, #19722 and #13939

### CODE REVIEW

File: docs/ru/sql-reference/aggregate-functions/parametric-functions.md [hunk 1]
```diff
@@ -239,7 +239,7 @@ windowFunnel(window, [mode])(timestamp, cond1, cond2, ..., condN)
 
 **Параметры**
 
--   `window` — ширина скользящего окна по времени в секундах. [UInt](../../sql-reference/aggregate-functions/parametric-functions.md).
+-   `window` — ширина скользящего окна по времени. Единица измерения зависит от `timestamp` и может варьироваться. Определяется выражением `timestamp от cond2 <= timestamp от cond1 + window`.
```
Comment (olgarev) [inline docs/ru/sql-reference/aggregate-functions/parametric-functions.md:242]:
```suggestion
-   `window` — ширина скользящего окна по времени. Единица измерения зависит от `timestamp` и может варьироваться. Должно соблюдаться условие `timestamp события cond2 <= timestamp события cond1 + window`.
```

File: docs/ru/sql-reference/functions/encryption-functions.md [hunk 2]
```diff
@@ -11,7 +11,7 @@ toc_title: "\u0424\u0443\u043d\u043a\u0446\u0438\u0438 \u0434\u043b\u044f \u0448
 
 Длина инициализирующего вектора всегда 16 байт (лишнии байты игнорируются). 
 
-Обратите внимание, что эти функции работают медленно.
+Обратите внимание, что до версии Clickhouse 21.1 эти функции работают медленно.
```
Comment (olgarev) [inline docs/ru/sql-reference/functions/encryption-functions.md:14]:
```suggestion
Обратите внимание, что до версии Clickhouse 21.1 эти функции работали медленно.
```

File: docs/ru/sql-reference/functions/encryption-functions.md [hunk 3]
```diff
@@ -52,99 +52,66 @@ encrypt('mode', 'plaintext', 'key' [, iv, aad])
...
+│ aes-256-gcm          │ A8A3CCBC6426CFEEB60E4EAE03D3E94204C1B09E0254 │
+│ aes-256-gcm with AAD │ A8A3CCBC6426D9A1017A0A932322F1852260A4AD6837 │
+└──────────────────────┴──────────────────────────────────────────────┘
 ```
 
 ## aes_encrypt_mysql {#aes_encrypt_mysql}
 
 Совместима с шифрованием myqsl, результат может быть расшифрован функцией [AES_DECRYPT](https://dev.mysql.com/doc/refman/8.0/en/encryption-functions.html#function_aes-decrypt).
 
+При одинаковых вводных зашифрованный текст будет совпадать с результатом `encrypt`. Однако, когда `key` или `iv` длиннее, чем должны быть, `aes_encrypt_mysql` будет работать аналогично MySQL `aes_encrypt`: свернет ключ и проигнорирует лишнюю часть `iv`.
```
Comment (olgarev) [inline docs/ru/sql-reference/functions/encryption-functions.md:113]:
```suggestion
При одинаковых входящих значениях зашифрованный текст будет совпадать с результатом, возвращаемым функцией `encrypt`. Однако если `key` или `iv` длиннее, чем должны быть, `aes_encrypt_mysql` будет работать аналогично MySQL `aes_encrypt`: свернет ключ и проигнорирует лишнюю часть `iv`.
```

File: docs/ru/sql-reference/functions/encryption-functions.md [hunk 4]
```diff
@@ -156,86 +123,104 @@ SELECT 'aes-192-gcm' AS mode, hex(encrypt(mode, input, key24, iv, 'AAD')) FROM e
...
 
 **Возвращаемое значение**
 
--   Зашифрованная строка. [String](../../sql-reference/data-types/string.md#string).
+-   Двоичная зашифрованная строка. [String](../../sql-reference/data-types/string.md#string).
 
 **Примеры**
 
-Создадим такую таблицу:
+При одинаковых вводных результаты шифрования `encrypt` и `aes_encrypt_mysql` будут совпадать.
```
Comment (olgarev) [inline docs/ru/sql-reference/functions/encryption-functions.md:143]:
```suggestion
При одинаковых входящих значениях результаты шифрования у функций `encrypt` и `aes_encrypt_mysql`  совпадают.
```

File: docs/ru/sql-reference/functions/encryption-functions.md [hunk 5]
```diff
@@ -156,86 +123,104 @@ SELECT 'aes-192-gcm' AS mode, hex(encrypt(mode, input, key24, iv, 'AAD')) FROM e
...
-Вставим эти данные:
+Результат:
+
+``` text
+┌─ciphertexts_equal─┐
+│                 1 │
+└───────────────────┘
+```
+
+Но `encrypt` генерирует исключение, когда `key` или `iv` длиннее, чем нужно:
```
Comment (olgarev) [inline docs/ru/sql-reference/functions/encryption-functions.md:159]:
```suggestion
Но `encrypt` генерирует исключение, если `key` или `iv` длиннее, чем нужно:
```

File: docs/ru/sql-reference/functions/encryption-functions.md [hunk 6]
```diff
@@ -156,86 +123,104 @@ SELECT 'aes-192-gcm' AS mode, hex(encrypt(mode, input, key24, iv, 'AAD')) FROM e
...
+
+Результат:
+
+``` text
+Received exception from server (version 21.1.2):
+Code: 36. DB::Exception: Received from localhost:9000. DB::Exception: Invalid key size: 33 expected 32: While processing encrypt('aes-256-cfb128', 'Secret', '123456789101213141516171819202122', 'iviviviviviviviv123'). 
 ```
 
-Пример без `iv`:
+Тогда как `aes_encrypt_mysql` возвращает совместимый с MySQL вывод:
```
Comment (olgarev) [inline docs/ru/sql-reference/functions/encryption-functions.md:174]:
```suggestion
Однако функция `aes_encrypt_mysql` возвращает совместимый с MySQL вывод:
```

File: docs/ru/sql-reference/functions/encryption-functions.md [hunk 7]
```diff
@@ -156,86 +123,104 @@ SELECT 'aes-192-gcm' AS mode, hex(encrypt(mode, input, key24, iv, 'AAD')) FROM e
...
++------------------------+
+| 0x24E9E4966469         |
++------------------------+
+1 row in set (0.00 sec)
 ```
 
 ## decrypt {#decrypt}
 
-Функция поддерживает расшифровку данных следующими режимами:
+Функция расшифровывает зашифрованный текст в обычный следующими режимами:
```
Comment (olgarev) [inline docs/ru/sql-reference/functions/encryption-functions.md:223]:
```suggestion
Функция расшифровывает зашифрованный текст и может работать в следующих режимах:
```

File: docs/ru/sql-reference/functions/encryption-functions.md [hunk 8]
```diff
@@ -156,86 +123,104 @@ SELECT 'aes-192-gcm' AS mode, hex(encrypt(mode, input, key24, iv, 'AAD')) FROM e
...
-│ aes-256-cfb128 │                                                            │
-│ aes-256-cfb128 │ 7FB039F7                                                   │
-│ aes-256-cfb128 │ 5CBD20F7ABD3AC41FCAA1A5C0E119E2BB5174F                     │
-└────────────────┴────────────────────────────────────────────────────────────┘
+┌─ciphertext───┐
+│ 24E9E4966469 │
+└──────────────┘
+```
+
+Это совпадает с тем, что выводит MySQL с такими же вводными:
```
Comment (olgarev) [inline docs/ru/sql-reference/functions/encryption-functions.md:206]:
```suggestion
Это совпадает с результатом, возвращаемым MySQL при таких же входящих значениях:
```

File: docs/ru/sql-reference/functions/encryption-functions.md [hunk 9]
```diff
@@ -265,51 +250,57 @@ decrypt('mode', 'ciphertext', 'key' [, iv, aad])
 
 **Примеры**
 
-Создадим такую таблицу:
+Используется таблица из [encrypt](#encrypt).
```
Comment (olgarev) [inline docs/ru/sql-reference/functions/encryption-functions.md:253]:
```suggestion
Рассмотрим таблицу из примера для функции [encrypt](#encrypt).
```

File: docs/ru/sql-reference/functions/encryption-functions.md [hunk 10]
```diff
@@ -265,51 +250,57 @@ decrypt('mode', 'ciphertext', 'key' [, iv, aad])
...
 ```
 
+Теперь попытаемся расшифровать эти данные:
+
 Запрос:
 
 ``` sql
-
-SELECT 'aes-128-ecb' AS mode, decrypt(mode, encrypt(mode, input, key16), key16) FROM encryption_test;
+SELECT comment, decrypt('aes-256-cfb128', secret, '12345678910121314151617181920212') as plaintext FROM encryption_test
```
Comment (olgarev) [inline docs/ru/sql-reference/functions/encryption-functions.md:281]:
```suggestion
SELECT comment, decrypt('aes-256-cfb128', secret, '12345678910121314151617181920212') as plaintext FROM encryption_test;
```

File: docs/ru/sql-reference/functions/encryption-functions.md [hunk 11]
```diff
@@ -265,51 +250,57 @@ decrypt('mode', 'ciphertext', 'key' [, iv, aad])
...
+┌─comment─────────────────────────────┬─plaintext─┐
+│ aes-256-cfb128 no IV                │ Secret    │
+│ aes-256-cfb128 no IV, different key │ �4�
+                                           �         │
+│ aes-256-cfb128 with IV              │ ���6�~        │
+ │aes-256-cbc no IV                   │ �2*4�h3c�4w��@
+└─────────────────────────────────────┴───────────┘
 ```
 
+Обратите внимание, что только часть данных была расшифрована, а остальное является бессмыслицей, как как `mode`, `key`, или `iv` были другими во время шифрования.
```
Comment (olgarev) [inline docs/ru/sql-reference/functions/encryption-functions.md:296]:
```suggestion
Обратите внимание, что только часть данных была расшифрована верно. Оставшаяся часть расшифрована некорректно, так как при шифровании использовались другие значения `mode`, `key`, или `iv`.
```

File: docs/ru/sql-reference/functions/encryption-functions.md [hunk 12]
```diff
@@ -265,51 +250,57 @@ decrypt('mode', 'ciphertext', 'key' [, iv, aad])
...
+
 ## aes_decrypt_mysql {#aes_decrypt_mysql}
 
 Совместима с шифрованием myqsl и может расшифровать данные, зашифрованные функцией [AES_ENCRYPT](https://dev.mysql.com/doc/refman/8.0/en/encryption-functions.html#function_aes-encrypt). 
 
+При одинаковых вводных расшифрованный текст будет совпадать с результатом `decrypt`. Однако, когда `key` или `iv` длиннее, чем должны быть, `aes_decrypt_mysql` будет работать аналогично MySQL `aes_decrypt`: свернет ключ и проигнорирует лишнюю часть `iv`.
```
Comment (olgarev) [inline docs/ru/sql-reference/functions/encryption-functions.md:302]:
```suggestion
При одинаковых входящих значениях расшифрованный текст будет совпадать с результатом, возвращаемым функцией `decrypt`. Однако если `key` или `iv` длиннее, чем должны быть, `aes_decrypt_mysql` будет работать аналогично функции `aes_decrypt` в MySQL: свернет ключ и проигнорирует лишнюю часть `iv`.
```

File: docs/ru/sql-reference/functions/encryption-functions.md [hunk 13]
```diff
@@ -265,51 +250,57 @@ decrypt('mode', 'ciphertext', 'key' [, iv, aad])
...
+
 Функция поддерживает расшифровку данных следующими режимами:
```
Comment (olgarev) [inline docs/ru/sql-reference/functions/encryption-functions.md:304]:
```suggestion
Функция поддерживает расшифровку данных в следующих режимах:
```

File: docs/ru/sql-reference/functions/encryption-functions.md [hunk 14]
```diff
@@ -332,51 +323,39 @@ aes_decrypt_mysql('mode', 'ciphertext', 'key' [, iv])
...
 **Возвращаемое значение**
 
 -   Расшифрованная строка. [String](../../sql-reference/data-types/string.md#string).
 
 **Примеры**
 
-Создадим такую таблицу:
-
-Запрос:
+Расшифруем данные, которые до этого зашифровали с помощью MySQL:
```
Comment (olgarev) [inline docs/ru/sql-reference/functions/encryption-functions.md:332]:
```suggestion
Расшифруем данные, которые до этого зашифровали в MySQL:
```

File: docs/ru/sql-reference/functions/encryption-functions.md [hunk 15]
```diff
@@ -332,51 +323,39 @@ aes_decrypt_mysql('mode', 'ciphertext', 'key' [, iv])
...
+| 0x24E9E4966469         |
++------------------------+
+1 row in set (0.00 sec)
 ```
 
 Запрос:
 
 ``` sql
-SELECT 'aes-128-cbc' AS mode, aes_decrypt_mysql(mode, aes_encrypt_mysql(mode, input, key), key) FROM encryption_test;
+SELECT aes_decrypt_mysql('aes-256-cfb128', unhex('24E9E4966469'), '123456789101213141516171819202122', 'iviviviviviviviv123456') AS plaintext
```
Comment (olgarev) [inline docs/ru/sql-reference/functions/encryption-functions.md:351]:
```suggestion
SELECT aes_decrypt_mysql('aes-256-cfb128', unhex('24E9E4966469'), '123456789101213141516171819202122', 'iviviviviviviviv123456') AS plaintext;
```

File: docs/ru/engines/table-engines/integrations/kafka.md [hunk 16]
```diff
@@ -31,21 +31,26 @@ SETTINGS
...
+-   `kafka_num_consumers` — количество потребителей (consumer) на таблицу. По умолчанию: `1`. Укажите больше потребителей, если пропускная способность одного потребителя недостаточна. Общее число потребителей не должно превышать количество партиций в топике, так как на одну партицию может быть назначено не более одного потребителя.
+-   `kafka_max_block_size` — максимальный размер пачек (в сообщениях) для poll (по умолчанию `max_block_size`).
+-   `kafka_skip_broken_messages` — максимальное количество некорректных сообщений в блоке. Если `kafka_skip_broken_messages = N`, то движок отбрасывает `N` сообщений Кафки, которые не получилось обработать. Одно сообщение в точности соответствует одной записи (строке). Значение по умолчанию – 0.
+-   `kafka_commit_every_batch` — фиксирует каждый обработанный и потребленный пакет вместо отдельной фиксации после записи целого блока (по умолчанию `0`).
+-   `kafka_thread_per_consumer` — снабжает каждого потребителя независимым потоком (по умолчанию `0`). При включенном состоянии каждый потребитель сбрасывает данные независимо и параллельно (иначе — строки от нескольких потребителей склеиваются в один блок).
```
Comment (olgarev) [inline docs/ru/engines/table-engines/integrations/kafka.md:53]:
```suggestion
-   `kafka_commit_every_batch` — включает или отключает режим записи каждой принятой и обработанной пачки по отдельности вместо единой записи целого блока (по умолчанию `0`).
-   `kafka_thread_per_consumer` — включает или отключает предоставление отдельного потока каждому потребителю (по умолчанию `0`). При включенном режиме каждый потребитель сбрасывает данные независимо и параллельно, при отключённом — строки с данными от нескольких потребителей собираются в один блок.
```

**Review by nikitamikhaylov — APPROVED** [review]

**Comment by olgarev** [comment]
@nikitamikhaylov Hi, is everything OK here? Anything left? The PR was not merged.

**Comment by nikitamikhaylov** [comment]
> @nikitamikhaylov Hi, is everything OK here? Anything left? The PR was not merged.

I was a bit confused about the translation of Kafka settings and the presence of some settings at all. So, there will be a bit of extra work from my side.

**Comment by gyuton** [comment]
@nikitamikhaylov Hi! Any additions yet?

**Pull request merged**